pypeit/spectrographs/p200_dbsp.py

```diff
@@ -0,0 +1,432 @@
+        #meta['airmass'] = dict(ext=0, card='AIRMASS')
+        # VERY HACKY!!!!
+        meta['airmass'] = dict(card=None, compound=True)
+
+        # Extras for config and frametyping
+        meta['dichroic'] = dict(ext=0, card='DICHROIC')
+        meta['dispangle'] = dict(card=None, rtol=1e-2, compound=True)
+        meta['slitwid'] = dict(ext=0, card='APERTURE')
+        meta['idname'] = dict(ext=0, card='IMGTYPE')
+        # Lamps
+        meta['lampstat01'] = dict(ext=0, card='LAMPS')
+
+        # Ingest
+        self.meta = meta
+
+    def compound_meta(self, headarr: List[fits.Header], meta_key):
+        if meta_key == 'mjd':
+            return Time(headarr[0]['UTSHUT']).mjd
+        elif meta_key == 'dispangle':
+            try:
+                return Angle(headarr[0]['ANGLE']).deg
+            except Exception as e:
+                print(headarr[0]['ANGLE'])
+                raise e
+        elif meta_key == 'ra':
+            ra = headarr[0].get('RA', default=None)
+            frametype = headarr[0]['IMGTYPE']
+            if ra is None and frametype in ['bias', 'flat', 'cal']:
+                ra = get_zenith_ra_dec(headarr[0]['UTSHUT']).ra.to_string(unit='hour', sep=':')
+            return ra
+        elif meta_key == 'dec':
+            dec = headarr[0].get('DEC', default=None)
+            frametype = headarr[0]['IMGTYPE']
+            if dec is None and frametype in ['bias', 'flat', 'cal']:
+                dec = get_zenith_ra_dec(headarr[0]['UTSHUT']).dec.to_string(unit='deg', sep=':')
+            return dec
+        elif meta_key == 'airmass':
+            am = headarr[0].get('AIRMASS', default=None)
+            frametype = headarr[0]['IMGTYPE']
+            if am is None and frametype in ['bias', 'flat', 'cal']:
+                am = '1.000'
+            elif headarr[0].get('RA', default=None) and headarr[0].get('DEC', default=None):
+                ra = headarr[0]['RA']
+                dec = headarr[0]['DEC']
+                altaz = SkyCoord(ra, dec, unit=(u.hour, u.deg)).transform_to(AltAz(obstime=headarr[0]['UTSHUT'], location=loc))
+                am = altaz.secz
+            return am
+        else:
+            return None
+            #msgs.error("Not ready for this compound meta")
+
+    def pypeit_file_keys(self):
+        pypeit_keys = super(P200DBSPSpectrograph, self).pypeit_file_keys()
+        pypeit_keys += ['slitwid']
+        return pypeit_keys
+
+
+class P200DBSPBlueSpectrograph(P200DBSPSpectrograph):
+    """
+    Child to handle P200/DBSP blue specific code
+    """
+    def __init__(self):
+        # Get it started
+        super(P200DBSPBlueSpectrograph, self).__init__()
+        self.spectrograph = 'p200_dbsp_blue'
+        self.camera = 'DBSPb'
+    
+    def compound_meta(self, headarr, meta_key):
+        retval = super(P200DBSPBlueSpectrograph, self).compound_meta(headarr, meta_key)
+        if retval is not None:
+            return retval
+        if meta_key == 'binning':
```

Comment:
this looks the same for the red camera.  If so, I suggest you push this up into the Parent class.

Reply:
the red and blue cameras have their spectral and spatial axes flipped, so I think the only difference is the order in which I am unpacking the `CCDSUM` header